src/http/request.rs

```diff
@@ -59,58 +60,13 @@ impl<'a, Params> Request<'a, Params> {
         self
     }
 
-    #[must_use]
-    pub fn body_ref(&self) -> Option<&[u8]> {
-        self.body.as_deref()
-    }
-
-    #[must_use]
-    pub fn body_mut(&mut self) -> Option<&mut [u8]> {
-        self.body.as_deref_mut()
-    }
-
-    #[must_use]
-    pub fn headers_ref(&self) -> &Option<Headers> {
-        &self.headers
-    }
-
-    #[must_use]
-    pub fn headers_mut(&mut self) -> &mut Option<Headers> {
-        &mut self.headers
-    }
-
-    #[must_use]
-    pub fn method_ref(&self) -> &LightMethod {
-        &self.method
-    }
-
-    #[must_use]
-    pub fn route_ref(&self) -> &Route<'_> {
-        &self.route
-    }
-}
-
-// Makes sure only methods that use the params actually need to bound on them.
-impl<'a, Params> Request<'a, Params>
-where
-    Params: IntoIterator<Item = (&'static str, String)>,
-    Params: std::fmt::Debug + Clone,
-{
-    pub fn params(mut self, params: Option<Params>) -> Self {
-        self.params = params;
+    /// # Panics
+    /// Panics if the length of the slice exceeds the value of `MAX_PARAMS`.
+    pub fn params(mut self, params: &[(&'static str, String)]) -> Self {
+        self.params = params.try_into().expect("number of params exceeded capacity");
```

Comment:
```suggestion
    pub fn params(mut self, params: [(&'static str, String); MAX_PARAMS]) -> Self {
        self.params = params;
```

Reply:
The motivation here is that the original argument was of type `Option<Vec<_>>`. Converting a vec into an array requires calling `.try_into().unwrap()`. I guess the more explicit version is a better tradeoff.